activation/atxdb.go

```diff
@@ -280,6 +287,26 @@ func (db *ActivationDb) GetAtx(id types.AtxId) (*types.ActivationTx, error) {
 	return atx, nil
 }
 
+func (db *ActivationDb) IsIdentityActive(edsig string, layer types.LayerID) (bool, error){
+	epoch := layer.GetEpoch(uint16(db.LayersPerEpoch))
+	nodeId, err := db.ids.GetIdentity(edsig)
+	if err != nil {
+		return false, err
+	}
+	ids, err := db.GetNodeAtxIds(nodeId)
+	if err != nil {
+		return false, err
+	}
+	if len(ids) == 0 {
+		return false, fmt.Errorf("no active IDs")
+	}
+	atx, err := db.GetAtx(ids[len(ids)-1])
+	if err != nil {
+		return false, nil
+	}
+	return atx.Valid && atx.LayerIdx.GetEpoch(uint16(db.LayersPerEpoch)) == epoch, err
```

Comment:
@antonlerner does "atx.Valid" mean syntactically valid?
@zalmen we need to understand what should be stored if an ATX is invalid